### PR TITLE
feat: apply strict typescript rules to workers

### PR DESCRIPTION
**Pull request by capJavert**
This is the one of the PRs in a series to bring our TypeScript types to comply with [strict](https://www.typescriptlang.org/tsconfig/#strict) config option. This will allow us better type checking during builds and development and allow us to ship with more confidence.

We are doing this as a series of PRs to have easier review and also to incrementally deploy to production.

This one focuses on:
- error type handling
- workers
  - especially notification workers and their types

### CODE REVIEW

File: __tests__/workers/notifications.ts [hunk 1]
```diff
@@ -158,7 +158,7 @@ it('should NOT add devcard unlocked notification if user had already reached the
       reputation: 40,
     },
   });
-  expect(actual).toEqual(null);
+  expect(actual).toEqual(undefined);
```
Comment (capJavert) [inline __tests__/workers/notifications.ts:161]: There was no point in returning `null` so now it's just `undefined`, worker expects falsy either way with `!value` check.

Comment (omBratteng) [reply __tests__/workers/notifications.ts:161]: I personally also use `toBeNull` for `null`, and `toBeUndefined` for `undefined`. Bit more explicit

Comment (capJavert) [reply __tests__/workers/notifications.ts]: Yeah, true

**Review by omBratteng — APPROVED** [review]

File: src/workers/commentDownvotedRep.ts [hunk 2]
```diff
@@ -27,6 +27,10 @@ const worker: TypedWorker<'api.v1.comment-downvoted'> = {
           .getRepository(User)
           .findOneBy({ id: data.userId });
 
+        if (!grantBy) {
+          return;
+        }
```
[the remark this answers is not in the document]
Comment (capJavert) [reply src/workers/commentDownvotedRep.ts:32]: Thought about it, but wanted to leave it to review. I am up for it, @omBratteng wdyt?

**Review by nensidosari — APPROVED** [review]

**Review by sshanzel — APPROVED** [review]
Nice work on this! :shipit:

File: src/workers/commentDownvotedRep.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (capJavert) [reply src/workers/commentDownvotedRep.ts]: Added in 6ed4ff88a22947669207798002c1ab21ef290031

**Pull request merged**